site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -54,6 +65,21 @@ public String save(Site site) {
         if (!validationResponse.isValid()) {
             throw new ValidationException(validationResponse.getErrorMessage());
         }
+
+        validationResponse = validateSiteExists(site);
+        if (!validationResponse.isValid()) {
+            throw new ValidationException(validationResponse.getErrorMessage());
+        }
+
         return siteStore.saveEntity(site);
     }
+
+    private ValidationResponse validateSiteExists(Site site) {
+        //Check if the Organization already exists.
+        Organization org = repository.findOrganizationByName(site.getName());
+        if (null != org) {
+            return new ValidationResponse(false, FhirRepo.SITE_EXISTS.message());
+        }
+        return new ValidationResponse(true, "");
+    }
```

Comment:
As we discussed yesterday, this is a validation operation that should exist in the SiteValidation class.
Or did you think something different?

Reply:
Yes I think something different

Reply:
Addressed as discussed with Tamir